src/Tribe/Views/V2/Views/Traits/iCal_Data.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Provides methods for getting iCal data for views
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views
+ */
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe\Events\Views\V2\View;
+
+/**
+ * Trait iCal_Data
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views\Traits
+ *
+ * @property string $string The slug of the View instance.
+ */
+trait iCal_Data {
+	/**
+	 * Returns the iCal data we're sending to the view.
+	 *
+	 * @since TBD
+	 *
+	 * @return object
+	 */
+	protected function get_ical_data() {
+		$slug = $this->slug;
+
+		/**
+		 * A filter to control whether the "iCal Import" link shows up or not.
+		 *
+		 * @since unknown
+		 *
+		 * @param boolean $show Whether to show the "iCal Import" link; defaults to true.
+		 */
+		$display_ical = apply_filters( 'tribe_events_list_show_ical_link', true );
+
+		/**
+		 * Allow for customization of the iCal export link "Export Events" text.
+		 *
+		 * @since unknown
+		 *
+		 * @param string $text The default link text, which is "Export Events".
+		 */
+		$link_text = apply_filters(
+			'tribe_events_ical_export_text',
+			sprintf(
+				/* translators: %s: Events (plural). */
+				__( 'Export %s', 'the-events-calendar' ),
+				tribe_get_event_label_plural()
+			)
+		);
+
+		$link_title = __( 'Use this to share calendar data with Google Calendar, Apple iCal and other compatible apps', 'the-events-calendar' );
+
+		switch( $slug ) {
+			case 'month':
+				$url = $this->get_month_view_url();
+				break;
+			default:
+				$url = tribe_get_ical_link();
+				break;
+		}
+
+		$ical_data = (object) [
+			'display_link' => $display_ical,
+			'link'         => (object) [
+				'url'   => esc_url( $url ),
+				'text'  => $link_text,
+				'title' => $link_title,
+			],
+		];
+
+		/**
+		 * Filters the ical data.
+		 *
+		 * @since 4.9.13
+		 *
+		 * @param object $ical_data An object containing the ical data.
+		 * @param View   $this      The current View instance being rendered.
+		 */
+		$ical_data = apply_filters( "tribe_events_views_v2_view_ical_data", $ical_data, $this );
+
```

Comment:
:no_entry_sign: String "tribe_events_views_v2_view_ical_data" does not require double quotes; use single quotes instead

<sub>hash: 35271fe6a5528cf3278cc20f964394cc</sub>